app/src/org/commcare/utils/StorageUtils.java

```diff
@@ -63,9 +62,10 @@ public static int getNumIncompleteForms() {
             return new FormRecord[0];
         }
 
-        // Order ids so they're submitted to and processed by the server in
-        // the correct order.
-        sortRecordsByDate(ids, storage);
+        if (sortRequired) {
+            // Order ids so they're submitted to and processed by the server in the correct order.
+            sortRecordsBySubmissionOrderingNumber(ids, storage);
```

Comment:
I mean why not fetch all the records first and then sort them? Currently there are 2 storage operations per form:
* get ordering number
* get form record

Just wondering if that could be reduced to 1 if the form records were fetched up front. 

Perhaps just my understanding of how the storage works is wrong.